packages/medusa/src/api-v2/store/carts/[id]/payment-collections/route.ts

```diff
@@ -0,0 +1,51 @@
+import { createPaymentCollectionForCartWorkflow } from "@medusajs/core-flows"
+import { UpdateCartDataDTO } from "@medusajs/types"
+import { remoteQueryObjectFromString } from "@medusajs/utils"
+
+import { MedusaRequest, MedusaResponse } from "../../../../../types/routing"
+import { defaultStoreCartFields } from "../../query-config"
+
+export const POST = async (
+  req: MedusaRequest<UpdateCartDataDTO>,
+  res: MedusaResponse
+) => {
+  const workflow = createPaymentCollectionForCartWorkflow(req.scope)
+
+  const remoteQuery = req.scope.resolve("remoteQuery")
+
+  let [cart] = await remoteQuery(
+    {
+      cart: {
+        fields: ["id", "currency_code", "region_id", "total"],
+      },
+    },
+    {
+      cart: { id: req.params.id },
+    }
+  )
+
+  const { errors } = await workflow.run({
+    input: {
+      cart_id: req.params.id,
+      region_id: cart.region_id,
+      currency_code: cart.currency_code,
+      amount: cart.total ?? 0, // TODO: This should be calculated from the cart when totals decoration is introduced
+    },
+    throwOnError: false,
```

Comment:
discussion: what do you think of having an onError callback handler instead of this configuration?
it would look like:

```ts
import { workflowApiErrorHandler } from '@medusajs/utils'
const { errors } = await workflow.run({
    input: {
      cart_id: req.params.id,
      region_id: cart.region_id,
      currency_code: cart.currency_code,
      amount: cart.total ?? 0, // TODO: This should be calculated from the cart when totals decoration is introduced
    },
   onError: workflowApiErrorHandler
})
```

```ts
export function workflowApiErrorHandler(errors) {
  if  (Array.isArray(errors) && errors[0]) {
    throw errors[0].error
  }
}
```

Reply:
I think I like this idea. It is def. better than what we do now. 

We could export good error handlers for when we use workflows in different places:
- Rest API -> workflowsApiRouteErrorHandler
- Modules- > workflowsModulesErrorHandler
- ...


However, we should still find a good way to throw an appropriate Medusa error in an API Route. Right now, if an error is thrown in a workflow executed in an API Route, I believe we respond with a 500 "unknown error", because our error handler doesn't know how to treat workflow errors.

Reply:
no in theory if you re throw the error it should be a medusa error if it was a medusa error thrown, you can check what the error is to be sure

Reply:
Let's punt this discussion for when we've built some more workflows. Think it would be good to have experienced some more cases before we proceed with a solution